refactor/3052 refactor to avoid having too many files in root

davidberenstein1957 opened the pull request:
# Description

Moved some stuff
- docker-related files to `docker` directory
- `scripts` directory to `docker` directory 
- general files to `.github` 

Closes #3052

**Type of change**

- [X] Refactor (change restructuring the codebase without changing functionality)

**How Has This Been Tested**

N.A.

**Checklist**

N.A.

davidberenstein1957 committed:
refactor: moved config files to .github

davidberenstein1957 committed:
refactor: add path correction package flow

davidberenstein1957 committed:
refactor: setup quickstart docker with correct path to .github

davidberenstein1957 committed:
refactor: moved quickstarts back to root

frascuchon commented on .github/MANIFEST.in:
This file is used on package build. Not sure if we can move it from root without changes in pyproject.tml.

davidberenstein1957 replied:
The build seems to run without warnings but I can check to make sure.

davidberenstein1957 replied:
`pyproject.tml` does not seem to reference `MANIFEST.in`

frascuchon commented on docs/_source/.readthedocs.yaml:
This change will affect the docs build. See [the docs](https://docs.readthedocs.io/en/stable/config-file/v2.html#configuration-file-v2), but this file must be placed in the root of the project.

davidberenstein1957 replied:
@frascuchon  We can reference it in a nested directory too:) see my comment above.

frascuchon replied:
It looks like an experimental feature. I would prefer to keep the default value We never know with rtd builds... ;-)

<img width="586" alt="Screenshot 2023-06-01 at 22 01 20" src="https://github.com/argilla-io/argilla/assets/2518789/da16bf44-1684-4abc-ba00-62e166a55ea0">

frascuchon commented on .github/.codecov.yml:
Same here, this file should placed in the root folder

davidberenstein1957 replied:
I believe this should work in .github too https://github.com/codecov/codecov-ruby/issues/24#issuecomment-644772335

davidberenstein1957 committed:
refactor: created separate docker directory

davidberenstein1957 committed:
fix: move manifest back to root

davidberenstein1957 committed:
fix: updated path to build distribution script

davidberenstein1957 committed:
docs: updated docker download URLs

davidberenstein1957 committed:
refactor: moved environment_dev.yml

davidberenstein1957 committed:
fix: moved dockerignore back to root and added references

davidberenstein1957 committed:
fix: updated dockerignore

davidberenstein1957 committed:
fix: revert merge

davidberenstein1957 committed:
docs: move manifest to root and remove duplicate dockers from .github

frascuchon committed:
Remore refs to commitlint hook installation

frascuchon commented on docker/docker-compose.yaml:
We should change also the refs to this file in docs https://docs.argilla.io/en/latest/getting_started/installation/deployments/docker_compose.html#launching-argilla-and-elasticsearch-with-docker-compose

frascuchon replied:
Also here https://docs.argilla.io/en/latest/getting_started/installation/deployments/from_development.html#install-from-develop

davidberenstein1957 replied:
Thanks, I forgot those.

frascuchon commented on docker/docker-compose.opensearch.yaml:
I think we have also references to these files in docs?

davidberenstein1957 replied:
Yes, I had already updated those and the ones from `docker-compose.elasticsearch.yml`, but forgot about the docker-compose.

davidberenstein1957 committed:
docs: updated references to docker/docker-compose.yml

frascuchon reviewed (commented):
Regarding this [doc section](https://docs.github.com/en/communities), I'm not sure if we should move all those files into the dot-github folder. 

It looks like this folder is used for templates and workflow definitions. It's a bit weird for me to find README.md or CHANGELOG.md there, instead of the root one. Even more for the conda environment `environment_dev.yml`  or the coverage setup (`.codecov.yml`)

I would say:

1. keep common repo files in the root folder (README.md CHANGELOG, coverage setup, conda env) 
2. Move al docker stuff into a docker folder and review the required scripts
3. Remove unused files as the commitlint.config.js file or some obsolete script (I can take a look here)

Also, I think we should point this PR to the `develop` branch, right?

davidberenstein1957 committed:
made issues templates a bit friendlier (#3108)

- added a call to action about contacting me or starting a PR
- added some default tags to bug and features
- added a separate issue for docs

davidberenstein1957 committed:
added issues template for ui and integrations (#3109)

- added UI/UX specific bug template
- added a python/deployment bug template
- added an integration issue template

davidberenstein1957 committed:
fix: updated comments Paco
chore: added CONTRIBUTING.md

davidberenstein1957 committed:
chore: remove .github ref from pyproject.toml

frascuchon committed:
Merge branch 'develop' into feature/3052-refactor-to-avoid-having-too-many-files-in-root

frascuchon commented on .github/workflows/package.yml:
```diff
@@ -270,13 +271,13 @@ jobs:
-            dockerfile: release.Dockerfile
-            readme: README.md
+            dockerfile: docker/release.Dockerfile
+            readme: .github/README.md
```
```suggestion
            readme: README.md
```

davidberenstein1957 committed:
Update .github/workflows/package.yml

Co-authored-by: Francisco Aranda <francis@argilla.io>

frascuchon committed:
Review docker and scripts (#3133)

# Description

The docker and script have been reviewed:

- The docker build context will be placed in the docker folder, so we
don't need to change anything inside the dockerfiles
- The unneeded scripts have been removed
- The specific docker script has been moved into docker/scripts
- The scripts used outside docker have been placed in the scripts
folder, otherwise, script execution may change. And not much sense to
put them inside the docker folder since they are used outside docker
images.
- Docs have been updated
- The `release.Dockerfile` has been renamed to `Dockerfile` (more
standard way for the principal docker image)

---------

Co-authored-by: davidberenstein1957 <david.m.berenstein@gmail.com>

frascuchon committed:
Merge branch 'develop' into feature/3052-refactor-to-avoid-having-too-many-files-in-root

frascuchon commented on docker/Dockerfile:
```diff
@@ -30,4 +30,4 @@ VOLUME $ARGILLA_HOME_PATH
 EXPOSE 6900
 
-CMD /bin/bash /start_argilla_server.sh
+CMD ["/start_argilla_server.sh"]
```
```suggestion
CMD /bin/bash /start_argilla_server.sh
```

frascuchon committed:
Update docker/Dockerfile

frascuchon reviewed (approved):
It's working. Let's wait until the release branch creation before merging it, just to not disturb the release workflow

frascuchon committed:
Merge branch 'develop' into feature/3052-refactor-to-avoid-having-too-many-files-in-root
